forte/data/data_store.py

```diff
@@ -225,6 +229,223 @@ def __init__(
+                If there are fields that have different orders in the current
+                class and the serialized object, it switches the order of
+                fields to match the current class.
+                If there are fields that appear in the current class, but not in
+                the serialized object, it handles those fields with
+                `accept_unknown_attribute`.
+                If there are fields that appear in the serialized object, but
+                not in the current class, it drops those fields.
+            suppress_warning: Boolean value indicating whether users want to
+                see warnings when it checks attributes. Only applicable when
+                `check_attribute` is set to True. If true, it will
+                log warnings when there are mismatched fields.
+            accept_unknown_attribute: Boolean value indicating whether users
+                want to fill fields that appear in the current class, but not
+                in the serialized object with none. Only applicable when
+                `check_attribute` is set to True. If false, it will raise
+                an `ValueError` if there are any contradictions in fields.
+
+        Raises:
+            ValueError: raised when
+                1. the serialized object has unknown fields, but
+                    `accept_unknown_attribute` is False.
+                2. the serialized object does not store attributes, but
+                    `check_attribute` is True.
+                3. the serialized object does not support json deserialization.
+                    We may change this error when we have other options for
+                    deserialization.
+
+        Returns:
+            An data store object deserialized from the string.
+        """
+
+        def check_fields(store):
+            """
+            A helper function that compares fields of the serialized object and
+            the current `DataStore` class.
+
+            Args:
+                store: The serialized object we want to check and process.
+
+            Returns:
+                The data store object after we process its fields.
+            """
+            for t, v in cls._type_attributes.items():
+                change_map = {}
+                contradict_loc = []
+                # find fields that appear in the current class, but not in
+                # the serialized objects,
+                # or fields that have different orders in the current class
+                # and the serialized objects.
+                diff = set(v["attributes"].items()) - set(
+                    store._type_attributes[t]["attributes"].items()
+                )
+                for f in diff:
+                    # if fields appear in both the current class and the
+                    # serialized objects but have different orders, switch
+                    # fields to match the order of the current class.
```

Comment:
1. How did you find the attributes in different orders here? The `diff` is calculated using `set`, which means `diff` will only contain the missing elements from `store`. How can order be identified?
2. Also, this cannot handle attributes that are in the `store` but not in `cls`, right?